File: 496-next-greater-element.rb
def next_greater_element(find_nums, nums)
    result = []
    hash = {}
    nums.each_with_index do |el, idx|
        hash[el] = idx
    end
    find_nums.each_with_index do |el, i|
        index = hash[el] + 1
        if index == nums.length
            result << -1
        else
            (index...nums.length).each do |j|
                if (nums[j] > find_nums[i])
                    result << nums[j]
                    break
                elsif j === nums.length - 1
                    result << -1
                end
            end
        end
    end
    result
end

File: 496-next-greater-element.js
var nextGreaterElement = function(findNums, nums) {
    let obj = {};
    let result = [];
    for (let i = 0; i < nums.length; i++) {
        obj[nums[i]] = i;
    }
    for (let i = 0; i < findNums.length; i++) {
        let index = obj[findNums[i]] + 1;
        if (index === nums.length) {
            result.push(-1);
        } else {
          for (let j = index; j < nums.length; j++) {
            if (nums[j] > findNums[i]) {
                result.push(nums[j]);
                break;
            } else if (j === nums.length - 1) {
                result.push(-1);
            }
          }
        }
    }
    return result;
};
